### PR TITLE
EWPP-3846: Refactor WISYWIG trait and context for CKEditor5.

### PR DESCRIPTION
**Pull request by nagyad**

### CODE REVIEW

**Review by sergepavle — APPROVED** [review]

File: composer.json [hunk 1]
```diff
@@ -11,15 +11,14 @@
         "drupal/core": "^10",
         "drupal/datetime_range_timezone": "^1.0.0-alpha3",
         "drupal/linkit": "^6.0.0-beta4",
-        "drupal/maxlength": "^2.0.1",
+        "drupal/maxlength": "dev-3.x",
```
Comment (imanoleguskiza) [inline composer.json]: I think it might be better to point to a specific commit instead of having dev

File: tests/src/Behat/WysiwygContext.php [hunk 2]
```diff
@@ -52,15 +52,16 @@ public function assertNoWysiwyg($label) {
...
    */
-  public function iShouldSeeWysiwygButton($field, $button) {
+  public function iShouldSeeWysiwygButton(string $field, string $button): void {
     $wysiwyg = $this->getWysiwyg($field);
-    $button_elements = $this->getSession()->getDriver()->find($wysiwyg->getXpath() . '//a[@title="' . $button . '"]');
-    if (empty($button_elements)) {
-      throw new \Exception("Could not find the '$button' button.");
-    }
-    if (count($button_elements) > 1) {
-      throw new \Exception("Multiple '$button' buttons found in the editor.");
```
Comment (imanoleguskiza) [inline tests/src/Behat/WysiwygContext.php:62]: Why did we lose this assertion?

**Review by imanoleguskiza — DISMISSED** [review]

File: composer.json (hunk 1 above)
Comment (nagyad) [reply composer.json]: It is temporary until we get the new release, in order to have dependencies handled correctly.

File: tests/src/Behat/WysiwygContext.php (hunk 2 above)
Comment (nagyad) [reply tests/src/Behat/WysiwygContext.php:62]: You can't have the same button multiple times in an editor. Config doesn't allow it.

**Review by sergepavle — DISMISSED** [review]

**Review by sergepavle — APPROVED** [review]

**Review by 22Alexandra — APPROVED** [review]
blind for @imanoleguskiza

**Pull request merged**